dashboard/app/models/user_level.rb

```diff
@@ -97,6 +98,11 @@ def handle_unsubmit
     end
   end
 
+  # clear the per-request cache when user levels are saved or updated
+  def clear_user_level_cache
+    user.clear_user_level_cache unless user.nil
+  end
```

Comment:
also, the reference to the `user` association will add an extra database dip (`SELECT  'users'.* FROM 'users' WHERE 'users'.'id' = [user_id]`) every time a UserLevel is created/updated.

(this query should be bypassed via Rails' automatic `inverse_of` inference, but doesn't seem to be working as expected..)